fix: keep dynamically linked files in slim images

SimenB opened the pull request:
Should be merged after #1197. I kept them separate as I don't know how to test that this works for arm.

Fixes #1195

tianon commented:
I tested this on `arm32v7`, and found I had to add a couple things before it worked -- I had to add `libatomic1` to the list of things explicitly installed in the `node` layer, then I had to add the `savedAptMark` behavior we'd discussed previously in the `yarn` layer, ala:

```dockerfile
RUN set -ex \
  && savedAptMark="$(apt-mark showmanual)" \
  && apt-get update && apt-get install -y ca-certificates curl wget gnupg dirmngr --no-install-recommends \
# ...
  && apt-mark auto '.*' > /dev/null \
  && { [ -z "$savedAptMark" ] || apt-mark manual $savedAptMark > /dev/null; } \
  && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
  # smoke test
  && yarn --version
```

Kosta-Github commented:
can this be somehow resolved, since right now both `13.x` and `12.x` images are not working:
```bash
$ docker run --rm node:13-slim --version
node: error while loading shared libraries: libatomic.so.1: cannot open shared object file: No such file or directory
```
and
```bash
$ docker run --rm node:12-slim --version
node: error while loading shared libraries: libatomic.so.1: cannot open shared object file: No such file or directory
```
The last working one seems to be `12.15-slim`:
```bash
 $ docker run --rm node:12.15-slim --version
v12.15.0
```

SimenB commented:
I've rebased this now.

@tianon thanks for testing!

I added what I _think_ is correct as a second commit now. Does that look right? Could you test again?

nschonni commented on 10/buster-slim/Dockerfile:
```diff
@@ -51,6 +59,7 @@ RUN ARCH= && dpkgArch="$(dpkg --print-architecture)" \
 ENV YARN_VERSION 1.21.1
 
 RUN set -ex \
+  && savedAptMark="$(apt-mark showmanual)"
```
Missing the trialling `\` here and the other occurances
```suggestion
  && savedAptMark="$(apt-mark showmanual)" \
```

SimenB replied:
woops, fixed. thanks!

tianon commented:
Tested `13/buster-slim` on `arm32v7` and it was successful! :+1:

SimenB commented:
Wonderful, thanks for checking!

@nodejs/docker PTAL 🙂

nschonni commented:
Looks like the Travis job failed on the deploy if someone can try and reset it https://travis-ci.com/nodejs/docker-node/jobs/288915013

SimenB commented:
Yeah, failed when gh had issues earlier. I was unable to open a manual pr as well. I'll restart it 🙂
